server/use_cases/add_volunteer_shifts.py

```diff
@@ -0,0 +1,64 @@
+"""
+This module contains the use case for adding work shifts.
+"""
+from domains.work_shift import WorkShift
+
+
+def shift_add_use_case(repo, new_shift, existing_shifts):
+    """
+    The function adds a work shift into the chosen database
+    after checking for overlaps with existing shifts.
+    """
+    if shift_already_exists(new_shift, existing_shifts):
+        return {"success": False,
+                "message": "You are signed up for another shift at this time"}
+    new_shift_dict = new_shift.to_dict()
+    repo.add(new_shift_dict)
+    shift_id = new_shift_dict["_id"]
+    new_shift.set_id(shift_id)
+    new_shift_dict= new_shift.to_dict()
+    return {"id":shift_id, "success": True,
+            "message": "Shift added successfully"}
+
+
+def shift_add_multiple_use_case(repo, work_shifts, user_id):
+    """
+    Adds multiple work shifts into the database after checking for overlap.
+    """
+    if not work_shifts:
+        return []
+    #add the Volunteer object in the repo and
+    #create get_shifts_for_volunteer function
+    #to get the value of the sign_up_shifts
+    existing_shifts = repo.get_shifts_for_volunteer(user_id)
```

Comment:
Currently, in mongorepo, it only uses the old WorkShift object. In the future, it will have both ShelterVolunteer and Work objects. Then they can create get_shifts_for_volunteer function to get the value of the sign_up_shifts field. This field will include all the existing shifts for that user. We cannot add that to the current mongorepo since it will crash the application.